### ISSUE TITLE
Support for hidden commands - Feature Request

### ISSUE THREAD

**Issue by alexd-siaras** [docopt/docopt#232]
First off, docopts is incredible. It has made my team's life so much easier when we are putting together command line applications.

One omission that it would be great to have is to have support for hidden options (options that do not show up in the usage, but are parsed anyways).  We have a number of scripts that have some debug options that we use in house, but are not ready for prime time, and we have hid those from the end user.

**Comment by jasonsydes** [comment]
I agree, docopt is great!  

We also have quite a few features built into our tools that are very helpful for development (e.g. caching the results of a pipeline), but those features either a) aren't ready for production or b) will always been hidden.  

+1 for hidden commands.

Thank you again for docopt!

**Comment by keleshev** [comment]
How about something like this?

```python
"""API.

Usage: prog command [options]

Options:
  -h, --help
  -f, --foo
  -b, --bar

* * *

Private API options:
  -p, --private

"""
from docopt import docopt, DocoptExit


private_api = __doc__
public_api, _, _ = __doc__.partition('* * *')

try:
    print(docopt(public_api))
except DocoptExit:
    print(docopt(private_api))
```






Then you can:


```
$ python tmp.py 
Usage: prog command [options]

$ python tmp.py command
{'--bar': False,
 '--foo': False,
 '--help': False,
 'command': True}

$ python tmp.py command --foo
{'--bar': False,
 '--foo': True,
 '--help': False,
 'command': True}

$ python tmp.py command --private
{'--bar': None,
 '--foo': False,
 '--help': False,
 '--private': True,
 'command': True}
```

**Comment by alexd-siaras** [comment]
Another option would be to use a syntax is used for default:

"""API.

Usage: prog command [options]

Options:
  -h, --help
  -f, --foo
  -b, --bar
  -p, --private  [render-option: hidden]
"""

**Comment by keleshev** [comment]
Eh, that would break WYSIWYG, which is core nature of docopt.

**Comment by alexd-siaras** [comment]
You have a better grasp of the nature of docopt more-so than I. The suggestion that you provided also also doesn't fall into a WYSIWYG view, as the text in the private section is not seen. I think that any solution provided by docopt would violate this core nature. However, it doesn't change the need for the requirement :)

As you mentioned, there are ways around it by manipulating the option strings. I guess I think that it could be beneficial to have a standardized way so the users of docopt do not have to implement their own filtering mechanism.

**Comment by BurntSushi** [comment]
> The suggestion that you provided also also doesn't fall into a WYSIWYG view, as the text in the private section is not seen.

Right, which is why it's not being suggested for inclusion into Docopt proper, but as an acceptable solution to your requirement. :-)

I don't think this should be part of Docopt, especially when there's such an easy work around. We should resist feature creep.

**Comment by keleshev** [comment]
Another workaround I thought about is to use `help=False`, and manually print the redacted help. If you want hidden options/commands in the Usage-section too, then you can catch `DocoptExit` and print a redacted usage.

Yeah, maybe we should have a page with "docopt recipes" or something that describes how to use docopt in rare cases like this one.
